src/mapping/query/functions.rs

```diff
@@ -114,29 +276,70 @@ impl Function for ToFloatFn {
                 Value::Bytes(_) => Conversion::Float.convert(v).map_err(|e| e.to_string()),
                 Value::Boolean(b) => Ok(Value::Float(if b { 1.0 } else { 0.0 })),
                 Value::Timestamp(t) => Ok(Value::Float(t.timestamp() as f64)),
-                Value::Null => Err("value is null".to_string()),
-                _ => Err("unable to convert array or object into float".to_string()),
+                _ => unexpected_type!(v),
             },
             Err(err) => Err(err),
         }
         .or_else(|err| match &self.default {
-            Some(v) => Ok(v.clone()),
+            Some(v) => v.execute(ctx),
             None => Err(err),
         })
     }
+
+    fn parameters() -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kinds: &[
+                    ValueKind::Integer,
+                    ValueKind::Float,
+                    ValueKind::Bytes,
+                    ValueKind::Boolean,
+                    ValueKind::Timestamp,
+                ],
+                required: true,
+            },
+            Parameter {
+                keyword: "default",
+                kinds: &[
+                    ValueKind::Integer,
+                    ValueKind::Float,
+                    ValueKind::Bytes,
+                    ValueKind::Boolean,
+                    ValueKind::Timestamp,
+                ],
+                required: false,
+            },
+        ]
+    }
+}
+
+impl TryFrom<ArgumentList> for ToFloatFn {
+    type Error = String;
+
+    fn try_from(mut arguments: ArgumentList) -> Result<Self> {
+        let query = arguments.required("value")?;
+        let default = arguments.optional("default");
+
+        Ok(Self { query, default })
+    }
 }
 
 //------------------------------------------------------------------------------
 
 #[derive(Debug)]
 pub(in crate::mapping) struct ToBooleanFn {
     query: Box<dyn Function>,
-    default: Option<Value>,
+    default: Option<Box<dyn Function>>,
 }
 
 impl ToBooleanFn {
+    #[cfg(test)]
     pub(in crate::mapping) fn new(query: Box<dyn Function>, default: Option<Value>) -> Self {
-        Self { query, default }
+        Self {
+            query,
+            default: default.map(|v| Box::new(Literal::from(v)) as _),
+        }
```

Comment:
If you want to save a couple of lines, consider:
```suggestion
        let default = default.map(|v| Box::new(Literal::from(v)) as _);
        Self { query, default }
```